keras_cv/layers/object_detection/roi_pool.py

```diff
@@ -0,0 +1,141 @@
+
+
+class ROIPooler(tf.keras.layers.Layer):
+    """
+    Pooling feature map of dynamic shape into region of interest (ROI) of fixed shape.
+
+    Mainly used in Region CNN (RCNN) networks. This works for a single-level
+    input feature map.
+
+    This layer splits the feature map into [target_size[0], target_size[1]] areas,
+    and performs max pooling for each area. The area coordinates will be quantized.
+
+    Args:
+        bounding_box_format: a case-insensitive string.
+            For detailed information on the supported format, see the
+            [KerasCV bounding box documentation](https://keras.io/api/keras_cv/bounding_box/formats/).
+        target_size: List or Tuple of 2 integers of the pooled shape
+        image_shape: List of Tuple of 3 integers, or `TensorShape` of the image shape.
+
+    Usage:
+    ```python
+    feature_map = tf.random.normal([2, 16, 16, 512])
+    roi_pooler = ROIPooler("yxyx", [7, 7], [224, 224, 3])
+    rois = tf.constant([[[15., 30., 25., 45.]], [[22., 1., 30., 32.]]])
+    pooled_feature_map = roi_pooler(feature_map, rois)
+    ```
+    """
+
+    def __init__(
+        self,
+        bounding_box_format,
+        target_size,
+        image_shape,
+        **kwargs,
+    ):
+        if not isinstance(target_size, (tuple, list)):
+            raise ValueError(
+                f"Expected `target_size` to be tuple or list, got {type(target_size)}"
+            )
+        if len(target_size) != 2:
+            raise ValueError(
+                f"Expected `target_size` to be size 2, got {len(target_size)}"
+            )
+        super().__init__(**kwargs)
+        self.bounding_box_format = bounding_box_format
+        self.target_height = target_size[0]
+        self.target_width = target_size[1]
+        self.image_shape = image_shape
+        self.built = True
+
+    def call(self, feature_map, rois):
+        """
+        Args:
+          feature_map: [batch_size, H, W, C] float Tensor, the feature map extracted from image.
+          rois: [batch_size, N, 4] float Tensor, the region of interests to be pooled.
+        Returns:
+          pooled_feature_map: [batch_size, N, target_size, C] float Tensor
+        """
+        # convert to relative format given feature map shape != image shape
+        rois = bounding_box.convert_format(
+            rois,
+            source=self.bounding_box_format,
+            target="rel_yxyx",
+            image_shape=self.image_shape,
+        )
+        pooled_feature_map = tf.vectorized_map(
+            self._pool_single_sample, (feature_map, rois)
+        )
+        return pooled_feature_map
+
+    def _pool_single_sample(self, args):
+        """
+        Args: tuple of
+          feature_map: [H, W, C] float Tensor
+          rois: [N, 4] float Tensor
+        Returns:
+          pooled_feature_map: [target_size, C] float Tensor
+        """
+        feature_map, rois = args
+        N = rois.get_shape().as_list()[0]
```

Comment:
throughout the codebase we've been preferring name like:

```
N => num_rois
H => height
W => width
```

This is in-line with the style of Keras core.  No particularly strong opinion but this has been the case so far.

Reply:
Sticking to that style then.
Done.